chainerx_cc/chainerx/routines/normalization.cc

```diff
@@ -117,43 +272,70 @@ Array BatchNorm(
         Scalar eps,
         Scalar decay,
         const OptionalAxes& axis) {
+    // Preprocess inputs.
     PreprocessBatchNormResult result = PreprocessBatchNorm(x, gamma, beta, running_mean, running_var, axis);
-    std::shared_ptr<BatchNormForwardBackward> fb =
-            x.device().GetBatchNormForwardBackward(result.mean, result.var, eps, decay, result.sorted_axis);
-
     const Array& gamma_reshaped = result.gamma;
     const Array& beta_reshaped = result.beta;
+    const Array& mean_reshaped = result.mean;
+    const Array& var_reshaped = result.var;
+    const Axes& sorted_axis = result.sorted_axis;
+
+    Device& device = x.device();
+
+    // Compute forward.
+    Array out{};
+    std::shared_ptr<BatchNormGradState> state{};
+    {
+        NoBackpropModeScope scope{};
+        std::tie(out, state) = device.backend().CallOp<BatchNormOp>(
+                x.AsGradStopped(),
+                gamma_reshaped.AsGradStopped(),
+                beta_reshaped.AsGradStopped(),
```

Comment:
As it's wrapped with `NoBackpropModeScope`, I guess `AsGradStopped` is not needed.

Reply:
Thanks, fixed.